grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -271,4 +271,68 @@ class CycleCountService {
         }
         return cycleCounts.collect { CycleCountDto.createFromCycleCountItems(it.cycleCountItems) }
     }
+
+    CycleCountDto submitCount(CycleCountSubmitCountCommand command) {
+        command.cycleCount.cycleCountItems.each { CycleCountItem cycleCountItem ->
+            updateCycleCountItemForSubmit(cycleCountItem, command.refreshQuantityOnHand, command.failOnOutdatedQuantity)
+            determineCycleCountItemStatus(cycleCountItem, command.requireRecountOnDiscrepancy)
+        }
+        if (command.cycleCount.status == CycleCountStatus.COUNTED) {
+            createCycleCountTransaction(command.cycleCount)
+        }
+        return CycleCountDto.createFromCycleCountItems(command.cycleCount.cycleCountItems)
+    }
+
+    void updateCycleCountItemForSubmit(CycleCountItem cycleCountItem, boolean refreshQuantityOnHand, boolean failOnOutdatedQuantity) {
+        Integer currentQuantityOnHand =
+            productAvailabilityService.getQuantityOnHandInBinLocation(cycleCountItem.inventoryItem, cycleCountItem.location)
+        if (refreshQuantityOnHand) {
+            cycleCountItem.quantityOnHand = currentQuantityOnHand
+        }
+        if (failOnOutdatedQuantity && cycleCountItem.quantityOnHand != currentQuantityOnHand) {
+            throw new IllegalArgumentException("Quantity on hand for a cycle count item is no longer up to date")
+        }
+    }
+
+    void determineCycleCountItemStatus(CycleCountItem cycleCountItem, boolean requireRecountOnDiscrepancy) {
+        if (cycleCountItem.quantityOnHand == cycleCountItem.quantityCounted) {
+            cycleCountItem.status = CycleCountItemStatus.COUNTED
+            return
+        }
+        if (!requireRecountOnDiscrepancy) {
+            cycleCountItem.status = CycleCountItemStatus.APPROVED
```

Comment:
in the docs it was said, that the item state should be set to `DISCREPANCY_ACCEPTED` - we don't have such status for an item, so I assumed it's equivalent to `APPROVED`?

Reply:
I think both this if condition and the previous one should have status READY_TO_REVIEW.

When I was originally designing this I was imagining that we wanted a status that meant "there's a discrepancy, but we don't care". But it looks like we don't need to be able to distinguish that case so if !requireRecountOnDiscrepancy, then the item should always get status READY_TO_REVIEW because counting is completed.